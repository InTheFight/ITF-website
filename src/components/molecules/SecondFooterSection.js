import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  min-height: 460px;
  overflow: hidden;
  width: 100%;
  position: relative;
`;

const ImageBackground = styled.img`
  width: 100%;
  position: absolute;
  top: -20;
  z-index: -9999;
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  padding-left: 150px;
  width: 800px;
`;

const Text = styled.p`
  font-family: 'urw-din';
  font-weight: bold;
  font-size: 80px;
  color: #faf5eb;
  text-transform: uppercase;
  margin-bottom: 20px;
`;

const SecondFooterSection = ({ text, image }) => (
  <Container>
    <ImageBackground src={image} />
    <Content>
      <Text>{text}</Text>
    </Content>
  </Container>
);

SecondFooterSection.propTypes = {
  text: PropTypes.string.isRequired,
  image: PropTypes.node.isRequired,
};

export default SecondFooterSection;
