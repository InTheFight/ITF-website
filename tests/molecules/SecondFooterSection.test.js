import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SecondFooterSection from '../../src/components/molecules/SecondFooterSection';

test('Displays the second footer section', () => {
  const footerSection = render(<SecondFooterSection text='Sweet test quote' author='Test McTesterson' image={'test'}/>);

  expect(screen.getByText('Sweet test quote')).toBeInTheDocument();
  expect(footerSection).toMatchSnapshot();
});
