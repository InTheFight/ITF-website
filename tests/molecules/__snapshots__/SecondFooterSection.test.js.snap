// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays the second footer section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx fMNest"
      >
        <img
          class="sc-gtssRu ceqrpH"
          src="test"
        />
        <div
          class="sc-dlnjPT Kohpy"
        >
          <p
            class="sc-hKFyIo evCOZT"
          >
            Sweet test quote
          </p>
          <p
            class="sc-eCApGN ibGkeX"
          >
            -
            Test McTesterson
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx fMNest"
    >
      <img
        class="sc-gtssRu ceqrpH"
        src="test"
      />
      <div
        class="sc-dlnjPT Kohpy"
      >
        <p
          class="sc-hKFyIo evCOZT"
        >
          Sweet test quote
        </p>
        <p
          class="sc-eCApGN ibGkeX"
        >
          -
          Test McTesterson
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
